### PR TITLE
[MAINTENANCE] Clean up of `RegexPatternStringParameterBuilder` tests to use unittests

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- Follow-up of PR #4426. 
- Adds 3 unittests to `RegexPatternStringParameterBuilder`. 
    - One with incorrect column specified --> `get_metrics()` will now return an error
    - One with 1 `candidate_regex` specified (fixed by #4426)
    - One with 2 `candidate_regex` specified (was working, but converted to unittest)
 - Changed a few basic tests for `RegexPatternStringParameterBuilder` to use `Mock` rather than Alice DataContext. 

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py [hunk 1]
```diff
@@ -372,9 +372,14 @@ def get_metrics(
                     metric_configuration.metric_value_kwargs_id
                 ] = attributed_resolved_metrics
 
-            attributed_resolved_metrics.add_resolved_metric(
-                value=resolved_metrics[metric_configuration.id]
-            )
+                resolved_metric_value = resolved_metrics.get(metric_configuration.id)
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py]: Could we annotate this please 🙇🏽

**Review by cdkini — APPROVED** [review]
Beautiful! LGTM.

File: great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py [hunk 2]
```diff
@@ -278,7 +278,7 @@ def get_metrics(
         Then, all "MetricConfiguration" objects, collected into list as container, are resolved simultaneously.
         """
 
-        # Fist: Gather "metric_domain_kwargs" (corresponding to "batch_ids").
+        # First: Gather "metric_domain_kwargs" (corresponding to "batch_ids").
```
Comment (alexsherstinsky) [inline great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py:281]: Good grief!  Thanks for that fix, @Shinnnyshinshin !

File: tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py [hunk 3]
```diff
@@ -277,28 +326,65 @@ def test_regex_pattern_string_parameter_builder_bobby_no_match(
...
+
+    parameter_container: ParameterContainer = ParameterContainer(parameter_nodes=None)
+    domain: Domain = Domain(
+        domain_type=MetricDomainTypes.COLUMN, domain_kwargs=metric_domain_kwargs
+    )
+    with pytest.raises(ge_exceptions.ProfilerExecutionError) as e:
+        regex_pattern_string_parameter_builder.build_parameters(
+            parameter_container=parameter_container, domain=domain
+        )
+    assert (
```
Comment (alexsherstinsky) [inline tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py:362]: If you are going to push another commit, can you please add a blank line above `assert`?  Thanks!

**Review by alexsherstinsky — APPROVED** [review]
LGTM -- @Shinnnyshinshin big ❤️ for this one!

File: great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py]: Done!

File: tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py (hunk 3 above)
Comment (Shinnnyshinshin) [reply tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py:362]: done!

**Comment by Shinnnyshinshin** [comment]
Fixing `db-integration` tests.

**Pull request merged**